core/trino-server-rpm/src/main/resources/dist/etc/init.d/trino

```diff
@@ -37,6 +37,16 @@ CONFIGURATION=(
     --server-log-file "${SERVER_LOG_FILE:-/var/log/trino/server.log}"
 )
 
+files_limit=131072
+if current_limit=$(ulimit -Hn) && [ "$files_limit" -lt "$current_limit" ]; then
+    files_limit=$current_limit
+fi
+if max_limit=$(cat /proc/sys/fs/nr_open) && [ "$files_limit" -gt "$max_limit" ]; then
+    files_limit=$max_limit
+fi
+ulimit -Hn "$files_limit"
```

Comment:
nit: When the current_limit is gt what we want, we should not invoke `ulimit` at all.
Avoid calling ulimit unnecessarily,  as it could, in theory, fail.

Reply:
We'd have to either:
1. conditionally call `ulimit -Hn` and always call `ulimit -Sn`
2. also read the current value of `ulimit -S` and conditionally call both `ulimit -Hn` and `ulimit -Sn`
3. always call `ulimit -n` - I just checked the manual, and when neither `-H` nor `-S` is specified, it sets both limits. 

To keep things simple, I'd assume `ulimit` is declarative and always call it. It's a bash built-in, so we don't even start another process here. I changed this to be a single call.